### PR TITLE
Add 0.8 guide

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: website/source/guides/upgrading/upgrade-to-0.8.0.html.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+there is a maintenance window.
+
+There are two reindex processes that need to happen during the upgrade, for two
+different indexes. One will happen automatically when the cluster is upgraded
+to 0.8.0. The other can be done in one of a few ways, as follows.
+
+### If Not Using Replication
+
+If not using replication, and if you ever plan on using replication, you should
+reindex the primary by issuing a write to `sys/replication/reindex`. This can
```
Comment (chrishoffman) [inline website/source/guides/upgrading/upgrade-to-0.8.0.html.md]: Maybe add a link to the API docs.

File: website/source/guides/upgrading/upgrade-to-0.8.0.html.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+be done either on 0.7.3 prior to upgrading or 0.8.0 after upgrading, but will
+be faster if performed prior to upgrading.
+
+### If Using Replication
+
+#### Option 1: Reindex the Primary, then Upgrade Secondaries
+
+The first option is to issue a write to `sys/replication/reindex` on the
```
Comment (chrishoffman) [inline website/source/guides/upgrading/upgrade-to-0.8.0.html.md]: If adding the link above, also add here.

File: website/source/guides/upgrading/upgrade-to-0.8.0.html.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+There are two reindex processes that need to happen during the upgrade, for two
+different indexes. One will happen automatically when the cluster is upgraded
+to 0.8.0. The other can be done in one of a few ways, as follows.
+
+### If Not Using Replication
+
+If not using replication, and if you ever plan on using replication, you should
+reindex the primary by issuing a write to `sys/replication/reindex`. This can
+be done either on 0.7.3 prior to upgrading or 0.8.0 after upgrading, but will
+be faster if performed prior to upgrading.
```
Comment (briankassouf) [inline website/source/guides/upgrading/upgrade-to-0.8.0.html.md]: This might not be necessary with the new `cleanMerkleTree` function

Comment (jefferai) [inline website/source/guides/upgrading/upgrade-to-0.8.0.html.md]: Oh, good point.

**Pull request merged**